tests/model_forms/models.py

```diff
@@ -543,3 +543,22 @@ class Meta:
                 violation_error_message="Price must be greater than zero.",
             ),
         ]
+
+
+class AttnameConstraintsModel(models.Model):
+    left = models.ForeignKey(
+        "self", related_name="+", null=True, on_delete=models.SET_NULL
+    )
+    right = models.ForeignKey(
+        "self", related_name="+", null=True, on_delete=models.SET_NULL
+    )
+
+    class Meta:
+        required_db_features = {"supports_table_check_constraints"}
```

Comment:
EDIT ~The same flag should be checked in `test_check_constraint_refs_excluded_field_attname`.~ Nvd